allennlp/data/tokenizers/tokenizer.py

```diff
@@ -47,6 +47,18 @@ def tokenize(self, text: str) -> List[Token]:
         """
         raise NotImplementedError
 
+    def tokenize_sentences(self, sentences: List[str]) -> List[Token]:
```

Comment:
Why does this need to be on the base class? Seems like it should be a private method on the `pretrained_transformer_tokenizer` class. It has confusing behaviour that is only desirable for tokenising text for a language model.

Reply:
If it is a private method and it is used somewhere in DatasetReader, it will mean that this DatasetReader will only work with this type of tokenizer.

Reply:
However, I am not against making it private again. It will still be useful in many use cases,
and as @matt-gardner pointed out, we can always add it higher up later if necessary.

Reply:
Yes - my point is simply that currently, if you change the tokenizer to one which is not a pretrained lm tokenizer, what happens is pretty undesirable (your two sentences are tokenised as though they are a single sentence). Is this something you can handle in a `DatasetReader` without detecting that you aren't using one of the LM specific tokenizers in the first place? 

We can wait for Matt's opinion, but i'd suggest removing this.

Reply:
@DeNeutoy do you mean removing it completely or just making it private?

Reply:
We talked about this internally at a review meeting today, and I agree with @DeNeutoy's strong counterexample about the generality of this approach.  You can't just change tokenizers to go from BERT-style encoding to something else, you'd have to add new `TextFields` and such.  So it doesn't really make sense to have the method on the super class as I recommended.  Sorry.  We should move it back to a (public) method just on `PretrainedTransformerTokenizer`.  And given that, we probably don't need to handle more than two sentences, and just giving a sentence pair method for now is fine.  Sorry for the extra work that I recommended.

Reply:
Done!